play-dl/Request/index.ts

```diff
@@ -39,7 +31,37 @@ export function request_stream(req_url: string, options: RequestOpts = { method:
     });
 }
 /**
- * Main module which play-dl uses to make a proxy or normal request
+ * Makes a request and follows redirects if necessary
+ * @param req_url URL to make https request to
+ * @param cookies_added Whether cookies were added or not
+ * @param options Request options for https request
+ * @returns A promise with the final response object
+ */
+function internalRequest(
+    req_url: string,
+    cookies_added: boolean,
+    options: RequestOpts = { method: 'GET' }
+): Promise<IncomingMessage> {
+    return new Promise(async (resolve, reject) => {
+        let res = await https_getter(req_url, options).catch((err: Error) => err);
+        if (res instanceof Error) {
+            reject(res);
+            return;
+        }
+        if (res.headers && res.headers['set-cookie'] && cookies_added) {
+            cookieHeaders(res.headers['set-cookie']);
+        }
+        if (Number(res.statusCode) >= 300 && Number(res.statusCode) < 400) {
```

Comment:
Move this and other `>=400` status line before `set-cookie` header line.

This will ensure that if the status code is not a proper code, then it will not update cookies.

Reply:
Line 54 and 56.